src/e2e-test/features/bigquery/source/BigQueryToBigQuery.feature

```diff
@@ -262,3 +262,246 @@ Feature: BigQuery source - Verification of BigQuery to BigQuery successful data
     Then Open and capture logs
     Then Verify the pipeline status is "Succeeded"
     Then Validate the values of records transferred to BQ sink is equal to the values from source BigQuery table
+
+
+  @BQ_UPSERT_SOURCE_TEST @BQ_UPSERT_SINK_TEST
+  Scenario: Verify scenario form BigQuery To to ensure that upsert operations are performed without updating the destination table
```

Comment:
Rephrase the scenario outline and use the dedupe word also in the scenario to make it more undestandable.